sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client_async.py

```diff
@@ -40,6 +41,7 @@
     from azure.core.pipeline import Pipeline
     from azure.core.pipeline.transport import HttpRequest
     from azure.core.configuration import Configuration
+T = Optional[Union[BearerTokenCredentialPolicy, SharedKeyCredentialPolicy]]
```

Comment:
Rename it to something more explicit, it gives the impression that it's a generic, but it's not. Example `OptionalStorageCredentials`